### DISCUSSION TITLE
flow: Unhandled Rejection (TypeError): Cannot convert undefined or null to object

### DISCUSSION THREAD

**Discussion by qwertie** [mobxjs/mobx#2946] [Q&A]
The error above is produced by code such as this:

~~~
class Thing  {
    constructor() {
        makeObservable<Thing>(this, {
            something: flow,
        });
    }

    something = flow(function*(this: Thing): Generator<Promise<string>, number, string> {
        return 0;
    });
}

export const thing = new Thing();

// elsewhere...
    await thing.something();
~~~

The error goes away if I comment out `something: flow`. Is that the correct solution?

Call stack: 
- hasProp src/utils/utils.ts:170
- storeAnnotation src/api/decorators.ts:22
- Promise.decorator src/api/decorators.ts:12
- onFulfilled src/api/flow.ts:67
- (anonymous function) src/api/flow.ts:103
- Thing.res src/api/flow.ts:59

**Comment by urugator** [comment]
Either remove the annotation or the explicit `flow` call in the field initializer.
